packages/perennial/contracts/periphery/CoordinatorDelegatable.sol

```diff
@@ -0,0 +1,122 @@
+     */
+    function updateMaintenance(IProduct product, UFixed18 newMaintenance) external onlyOwnerOrParamAdmin {
+        product.updateMaintenance(newMaintenance);
+    }
+
+    /**
+     * @notice Updates the maker fee for product `product` to `newMakerFee`
+     * @dev Only callable by owner or paramAdmin
+     * @param product The product to update
+     * @param newMakerFee The new maker fee
+     */
+    function updateMakerFee(IProduct product, UFixed18 newMakerFee) external onlyOwnerOrParamAdmin {
+        product.updateMakerFee(newMakerFee);
+    }
+
+    /**
+     * @notice Updates the taker fee for product `product` to `newTakerFee`
+     * @dev Only callable by owner or paramAdmin
+     * @param product The product to update
+     * @param newTakerFee The new taker fee
+     */
+    function updateTakerFee(IProduct product, UFixed18 newTakerFee) external onlyOwnerOrParamAdmin {
+        product.updateTakerFee(newTakerFee);
+    }
+
+    /**
+     * @notice Updates the maker limit for product `product` to `newMakerLimit`
+     * @dev Only callable by owner or paramAdmin
+     * @param product The product to update
+     * @param newMakerLimit The new maker limit
+     */
+    function updateMakerLimit(IProduct product, UFixed18 newMakerLimit) external onlyOwnerOrParamAdmin {
+        product.updateMakerLimit(newMakerLimit);
+    }
+
+    /**
+     * @notice Updates the utilization curve for product `product` to `newUtilizationCurve`
+     * @dev Only callable by owner or paramAdmin
+     * @param product The product to update
+     * @param newUtilizationCurve The new utilization curve
+     */
+    function updateUtilizationCurve(IProduct product, JumpRateUtilizationCurve memory newUtilizationCurve) external onlyOwnerOrParamAdmin {
+        product.updateUtilizationCurve(newUtilizationCurve);
+    }
+
+    /**
+     * @notice Updates the param admin to `newParamAdmin`
+     * @dev Only callable by the owner
+     * @param newParamAdmin address of the new param admin
+     */
+    function updateParamAdmin(address newParamAdmin) external onlyOwner {
+        _paramAdmin.store(newParamAdmin);
+
+        emit CoordinatorDelegatableParamAdminUpdated(newParamAdmin);
+    }
+
+    /**
+     * @notice Executes an arbitrary function call or value transfer
+     * @dev Only callable by the owner
+     * @param to The target address
+     * @param data The calldata
+     * @param value The value to transfer
+     */
+    function execute(
+        address payable to,
+        bytes memory data,
+        uint256 value
+    ) payable external onlyOwner returns (bytes memory ret) {
+        if (data.length == 0) {
+            Address.sendValue(to, value);
+        } else {
+            ret = Address.functionCallWithValue(to, data, value);
+        }
+    }
+
+    modifier onlyOwnerOrParamAdmin {
+        if (_sender() != owner() && _sender() != paramAdmin())
+            revert CoordinatorDelegatableNotParamAdminError(_sender());
```

Comment:
nit: I've kind of stopped passing params back in these errors in favor of just debugging it like normal so I'm not sure if we want to still do that.

Reply:
I'll keep it since it's consistent with UOwnable for now